File: Programmers/소수 만들기.js
function solution(nums) {
  var answer = 0;
  const prime = [];
  const notPrime = [];

  for (let i = 0; i < nums.length - 2; i++) {
    for (let j = i + 1; j < nums.length - 1; j++) {
      for (let k = j + 1; k < nums.length; k++) {
        const target = nums[i] + nums[j] + nums[k];
        if (prime.includes(target)) answer++;
        else if (notPrime.includes(target)) continue;
        else {
          if (isPrime(target)) {
            prime.push(target);
            answer++;
            continue;
          }
          notPrime.push(target);
        }
      }
    }
  }

  return answer;
}

const isPrime = (n) => {
  let i = 2;
  while (i < n) {
    if (!(n % i)) return false;
    i++;
  }
  return true;
};

// const nums = [1, 2, 3, 4];
// // 1

const nums = [1, 2, 7, 6, 4];
// 4

console.log(solution(nums));

File: Programmers/소수 만들기.swift
//https://programmers.co.kr/learn/courses/30/lessons/12977?language=swift
//소수 만들기

import Foundation

func solution(_ nums:[Int]) -> Int {
    var answer = 0
    
    for i in 0..<nums.count-2 {
        for j in i+1..<nums.count-1 {
            for k in j+1..<nums.count {
                if isPrimeNumber(num: nums[i] + nums[j] + nums[k]) { answer += 1 }
            }
        }
    }
    
    return answer
}

func isPrimeNumber(num: Int) -> Bool {
    for i in 2..<num {
        if num % i == 0 { return false }
    }
    return true
}


//let nums: [Int] = [1,2,3,4]
////1

let nums: [Int] = [1,2,7,6,4]
//4

print(solution(nums))
